pkg/controller/config/config.go

```diff
@@ -140,3 +144,52 @@ func findCycle(path []SecretLocation, nodes map[SecretLocation]bool, edges map[S
 	}
 	return nil, false
 }
+
+// Load loads and parses the config at path.
+func Load(configLocation string) (c *Configuration, err error) {
+	// we never want config loading to take down the controller
+	defer func() {
+		if r := recover(); r != nil {
+			c, err = nil, fmt.Errorf("panic loading config: %v", r)
+		}
+	}()
+	c, err = loadConfig(configLocation)
+	if err != nil {
+		return nil, err
+	}
+	if err := c.Validate(); err != nil {
```

Comment:
we can just return `c, err` here (IMO it's fine to return a non-nil `Configuration` with a non-nil error, the caller is not supposed to use that value in that case anyway).

Reply:
OK. This one i would like to keep since it changes the return (although i agree that the `c` should not be used when `err` is not `nil`).